docs/internals/task_lifecycle.md

```diff
@@ -0,0 +1,186 @@
+---
+title:  "Task Lifecycle"
+nav-title: Task Lifecycle
+nav-parent_id: internals
+nav-pos: 5
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+A task in Flink is the basic unit of execution. It is the place where each parallel instance of an operator is executed. 
```

Comment:
"A task ... is the place where each parallel instance...": one could interpret that all parallel instances run in one task, which is misleading.